### PR TITLE
Update "Configuring Plugins" doc to cover project accessors and the plugins {} block

### PR DESCRIPTION
**Pull request by eskatos**

### CODE REVIEW

File: doc/getting-started/Configuring-Plugins.md [hunk 1]
```diff
@@ -33,10 +33,101 @@ greeting {
...
-The above groovy block of code would now become:
+```kotlin
+plugins {
+    // Gradle built-in
+    `application`
+    // From the Gradle Plugin Portal
+    id("com.bmuschko.docker-java-application") version "3.1.0"
+}
+
+// Type-safe accessor for the extension contributed by the `java` plugin
```
Comment (bamboo) [inline doc/getting-started/Configuring-Plugins.md]: :x: Contributed by the `application` plugin, right?

File: doc/getting-started/Configuring-Plugins.md [hunk 2]
```diff
@@ -33,10 +33,101 @@ greeting {
...
+        eachPlugin {
+            if (requested.id.id == "com.android.application") {
+                useModule("com.android.tools.build:gradle:${requested.version}")
+            }
+        }
+    }    
+}
+```
+
+You can now apply the `com.android.application` plugin using the `plugins {}` block and benefit from the type-safe
```
Comment (bamboo) [inline doc/getting-started/Configuring-Plugins.md]: 🎉

File: doc/getting-started/Configuring-Plugins.md [hunk 3]
```diff
@@ -33,10 +33,101 @@ greeting {
...
+}
+```
+
+See the [Plugin Management](https://docs.gradle.org/current/userguide/plugins.html#sec:plugin_management) section of
+the Gradle documentation for more information.
+
+However, it is not yet possible to use the `plugins {}` block in the following situations:
+- plugins already applied to a parent project
+- plugins from `buildSrc`
+- plugins from composite builds
```
Comment (eskatos) [inline doc/getting-started/Configuring-Plugins.md]: :x: should link to issues

**Comment by eskatos** [comment]
Thanks @bamboo, fixed, could you have another look?

**Pull request merged**